### PR TITLE
[16.0][MIG] helpdesk_mgmt_rating: Migration to 16.0

### PR DESCRIPTION
**Pull request by carolinafernandez-tecnativa**
Superseed https://github.com/OCA/helpdesk/pull/484

Continue Migration to 16.0

Changes done:

 - Squash administrative commits.
 - Fix issues from migration
 - Add new tests and improve test code coverage

@Tecnativa
TT48237

@pedrobaeza @victoralmau

### CODE REVIEW

File: helpdesk_mgmt_rating/migrations/16.0.1.0.0/post-migration.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.load_data(
+        env.cr, "helpdesk_mgmt_rating", "migrations/15.0.1.0.0/noupdate_changes.xml"
```
Comment (victoralmau) [inline helpdesk_mgmt_rating/migrations/16.0.1.0.0/post-migration.py]:
```suggestion
        env.cr, "helpdesk_mgmt_rating", "migrations/16.0.1.0.0/noupdate_changes.xml"
```

File: helpdesk_mgmt_rating/migrations/16.0.1.0.0/noupdate_changes.xml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        <field name="model_id" ref="helpdesk_mgmt.model_helpdesk_ticket" />
+        <field
+            name="email_from"
+        >{{object._rating_get_operator().email_formatted}}</field>
+        <field name="email_to">{{object._rating_get_partner().email_formatted}}</field>
+        <field
+            name="subject"
+        >Satisfaction Survey of the ticket {{object.number}} - {{object.name}}</field>
+        <field name="partner_to">{{object._rating_get_partner().id}}</field>
+        <field name="auto_delete" eval="True" />
```
Comment (victoralmau) [inline helpdesk_mgmt_rating/migrations/16.0.1.0.0/noupdate_changes.xml]:
```suggestion
        <field
            name="email_from"
        >{{object._rating_get_operator().email_formatted}}</field>
        <field name="email_to">{{object._rating_get_partner().email_formatted}}</field>
        <field name="partner_to">{{object._rating_get_partner().id}}</field>
```
Only what really changes should be added

File: helpdesk_mgmt_rating/tests/test_helpdesk_mgmt_rating.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            "mail_create_nosubscribe": True,
+            "mail_notrack": True,
+            "no_reset_password": True,
+        }
+        new_test_user(
+            cls.env,
+            login="test-helpdesk-user",
+            groups="helpdesk_mgmt.group_helpdesk_user",
+            context=ctx,
+        )
```
Comment (victoralmau) [inline helpdesk_mgmt_rating/tests/test_helpdesk_mgmt_rating.py]:
```suggestion
from odoo.tests.common import users
from odoo.addons.base.tests.common import BaseCommon


class TestHelpdeskMgmtRating(BaseCommon):
    @classmethod
    def setUpClass(cls):
        super().setUpClass()
        cls.partner = cls.env["res.partner"].create(
            {"name": "Test partner", "email": "test@email.com"}
        )
        new_test_user(
            cls.env,
            login="test-helpdesk-user",
            groups="helpdesk_mgmt.group_helpdesk_user",
        )
```

**Review by victoralmau — CHANGES_REQUESTED** [review]
Can you add some test to check that the controllers work (urls when clicking on the rating email icons) and it can be rating?

File: helpdesk_mgmt_rating/models/helpdesk_ticket.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                        force_send=force_send,
+                    )
+
+    def rating_apply(self, rate, token=None, feedback=None, subtype_xmlid=None):
+        return super().rating_apply(
+            rate,
+            token=token,
+            feedback=feedback,
+            subtype_xmlid="helpdesk_mgmt_rating.mt_ticket_rating",
+        )
```
Comment (victoralmau) [inline helpdesk_mgmt_rating/models/helpdesk_ticket.py]:
```suggestion
    def rating_apply(self, rate, token=None, rating=None, feedback=None,
                     subtype_xmlid=None, notify_delay_send=False):
        return super().rating_apply(
            rate,
            token=token,
            rating=rating,
            feedback=feedback,
            subtype_xmlid="helpdesk_mgmt_rating.mt_ticket_rating",
            notify_delay_send=notify_delay_send,
        )
```

**Review by victoralmau — CHANGES_REQUESTED** [review]

**Comment by carolinafernandez-tecnativa** [comment]
> Can you add some test to check that the controllers work (urls when clicking on the rating email icons) and it can be rating?

do you have an example?

**Comment by victoralmau** [comment]
> > Can you add some test to check that the controllers work (urls when clicking on the rating email icons) and it can be rating?
> 
> do you have an example?

Example: https://github.com/OCA/helpdesk/blob/16.0/helpdesk_mgmt/tests/test_helpdesk_portal.py#L108 The rating route must be set: /rate/{access_token}/5

File: helpdesk_mgmt_rating/migrations/16.0.1.0.0/noupdate_changes.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<?xml version='1.0' encoding='utf-8' ?>
+<odoo noupdate="1">
```
Comment (victoralmau) [inline helpdesk_mgmt_rating/migrations/16.0.1.0.0/noupdate_changes.xml]:
```suggestion
<odoo>
```

File: helpdesk_mgmt_rating/migrations/16.0.1.0.0/noupdate_changes.xml [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+    <record id="rating_ticket_email_template" model="mail.template">
+        <field
+            name="email_from"
+        >{{object._rating_get_operator().email_formatted}}</field>
+        <field name="email_to">{{object._rating_get_partner().email_formatted}}</field>
+        <field name="partner_to">{{object._rating_get_partner().id}}</field>
```
Comment (victoralmau) [inline helpdesk_mgmt_rating/migrations/16.0.1.0.0/noupdate_changes.xml:8]: You need to add `body_html` because it has been changed since version 15.0.

**Review by victoralmau — CHANGES_REQUESTED** [review]

**Comment by carolinafernandez-tecnativa** [comment]
> > > Can you add some test to check that the controllers work (urls when clicking on the rating email icons) and it can be rating?
> > 
> > 
> > do you have an example?
> 
> Example: https://github.com/OCA/helpdesk/blob/16.0/helpdesk_mgmt/tests/test_helpdesk_portal.py#L108 The rating route must be set: /rate/{access_token}/5

Added new tests, could you please check it now? Thanks!

**Review by victoralmau — APPROVED** [review]

**Comment by carolinafernandez-tecnativa** [comment]
ping @pedrobaeza

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**